src/api/app/controllers/webui/users/bs_requests_controller.rb

```diff
@@ -44,9 +44,9 @@ def filter_by_direction(direction)
         when 'all'
           User.session.requests
         when 'incoming'
-          User.session.incoming_requests
+          User.session.incoming_requests(states: BsRequest::VALID_REQUEST_STATES)
```

Comment:
```suggestion
          User.session.incoming_requests.unscope(where: :states)
```
I find it more readable if we use [unscope](https://api.rubyonrails.org/v7.0.8.1/classes/ActiveRecord/QueryMethods.html#method-i-unscope)...

Reply:
yeah but it's not scoping the requests, it's a method...

Reply:
Sure, it is a method which returns an instance of `BsRequest::ActiveRecord_Relation`. In fact, it is a method which scopes the requests...

I'm not sure why you prefer that the queries get populated with this conditionals in he `WHERE` section:
```
`bs_requests`.`state` IN ('new', 'deleted', 'declined', 'accepted', 'review', 'revoked', 'superseded')
```
... instead of removing the `state` condition...

Reply:
```
[5] pry(main)> unscope = User.find_by(login: 'hennevogel').incoming_requests.unscope(where: :states).to_sql
=> "SELECT DISTINCT `bs_requests`.* FROM `bs_requests` INNER JOIN `bs_request_actions` ON `bs_request_actions`.`bs_request_id` = `bs_requests`.`id` WHERE `bs_requests`.`state` = 'new' AND (`bs_requests`.`id` IN (SELECT `bs_request_actions`.`bs_request_id` FROM `bs_request_actions` WHERE `bs_request_actions`.`target_project_id` IN (65353, 65350, 269, 258616, 267647, 915698, 896257, 112979, 988347, 989382, 995871, 1003996, 1023344, 1062791, 1129417, 1138047, 1141595, 1142894, 1149071, 1153511, 1157702, 1161257, 1164542, 1299062, 1300718, 1307622, 1346458, 1357096, 1365260, 1375064, 1395569, 1395614, 1399094, 1399109, 1399460)) OR `bs_requests`.`id` IN (SELECT `bs_request_actions`.`bs_request_id` FROM `bs_request_actions` WHERE `bs_request_actions`.`target_package_id` IN (5340, 126711, 247363, 4533233, 9903507, 9903513, 9904974))) ORDER BY `bs_requests`.`priority` ASC, `bs_requests`.`id` DESC"
[6] pry(main)> scope = User.find_by(login: 'hennevogel').incoming_requests.to_sql
=> "SELECT DISTINCT `bs_requests`.* FROM `bs_requests` INNER JOIN `bs_request_actions` ON `bs_request_actions`.`bs_request_id` = `bs_requests`.`id` WHERE `bs_requests`.`state` = 'new' AND (`bs_requests`.`id` IN (SELECT `bs_request_actions`.`bs_request_id` FROM `bs_request_actions` WHERE `bs_request_actions`.`target_project_id` IN (65353, 65350, 269, 258616, 267647, 915698, 896257, 112979, 988347, 989382, 995871, 1003996, 1023344, 1062791, 1129417, 1138047, 1141595, 1142894, 1149071, 1153511, 1157702, 1161257, 1164542, 1299062, 1300718, 1307622, 1346458, 1357096, 1365260, 1375064, 1395569, 1395614, 1399094, 1399109, 1399460)) OR `bs_requests`.`id` IN (SELECT `bs_request_actions`.`bs_request_id` FROM `bs_request_actions` WHERE `bs_request_actions`.`target_package_id` IN (5340, 126711, 247363, 4533233, 9903507, 9903513, 9904974))) ORDER BY `bs_requests`.`priority` ASC, `bs_requests`.`id` DESC"
[7] pry(main)> scope == unscope
=> true
```

Reply:
Typo: `where: :states` should be `where: :state`, without the last `s`.